doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,597 @@
+the output will be equal to the number of steps we need to make, plus one,
+because we need to account for the initial position of the kernel. The same
+applies for the y-axis.
+
+From this, we conclude that the output size :math:`o` of the strided convolution
+is
+
+.. math::
+
+    o = \left\lfloor \frac{i - k}{s} \right\rfloor + 1.
+
+Combining padding and strides
+-----------------------------
+
+The most general case, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p`,
+
+is obtained by applying both the rule we derived for strides and the one we
+derived for the padding. In other words,
+
+.. math::
+
+    o = \left\lfloor \frac{i + 2p - k}{s} \right\rfloor + 1.
+
+Here is what it looks like for :math:`i = 5`, :math:`k = 3`, :math:`s = 2` and
+:math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides.gif
+
+Sometimes numbers are mean
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You may have noticed that the floor function in the general case formula will
+produce the same output shape for multiple input shapes. More specifically, if
+:math:`i + 2p - k` is a multiple of :math:`s`, then any input size
+:math:`j = i + a, \quad a \in \{0,\ldots,s - 1\}` will produce the same output
+size. Note that this ambiguity applies only for :math:`s > 1`.
+
+As an example, here is what a convolution looks like for :math:`i = 6`,
+:math:`k = 3`, :math:`s = 2` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/padding_strides_odd.gif
+
+Despite having different input sizes, this convolution and the previous
+convolution share the same output size.
+
+This may seem like a curiosity, but keep that in mind, as it will become
+important when we examine *transposed convolutions*.
+
+Transposed convolution: convolution turned on its head
+======================================================
+
+Every convolution boils down to an efficient implementation of a matrix
+operation. Take for instance the very first convolution presented in this
+tutorial. If the input and output were to be unrolled into vectors from left to
+right, top to bottom, the matrix representation of the convolution would look
+like
+
+.. math::
+
+    \begin{pmatrix}
+    w_{0,0} & 0       & 0       & 0       \\
+    w_{0,1} & w_{0,0} & 0       & 0       \\
+    w_{0,2} & w_{0,1} & 0       & 0       \\
+    0       & w_{0,2} & 0       & 0       \\
+    w_{1,0} & 0       & w_{0,0} & 0       \\
+    w_{1,1} & w_{1,0} & w_{0,1} & w_{0,0} \\
+    w_{1,2} & w_{1,1} & w_{0,2} & w_{0,1} \\
+    0       & w_{1,2} & 0       & w_{0,2} \\
+    w_{2,0} & 0       & w_{1,0} & 0       \\
+    w_{2,1} & w_{2,0} & w_{1,1} & w_{1,0} \\
+    w_{2,2} & w_{2,1} & w_{1,2} & w_{1,1} \\
+    0       & w_{2,2} & 0       & w_{1,2} \\
+    0       & 0       & w_{2,0} & 0       \\
+    0       & 0       & w_{2,1} & w_{2,0} \\
+    0       & 0       & w_{2,2} & w_{2,1} \\
+    0       & 0       & 0       & w_{2,2} \\
+    \end{pmatrix}^T
```

Comment:
Disregard my previous note. If possible, I would not transpose the matrix (it should fit in the page with the new template, which hopefully will be merged tomorrow)